### PR TITLE
Add collimate algorithm and Otsu improvements to OCR stitch block

### PR DESCRIPTION
**Pull request by reiffd7**
## What does this PR do?

Adds two new stitching algorithms to the OCR stitch block: "otsu" for automatic word boundary detection and "collimate" for handling skewed/curved text. 

- **Otsu algorithm**: Uses Otsu's method on normalized gap distances to automatically find the optimal threshold separating character gaps from word gaps. Resolution-invariant with bimodality detection to distinguish single words from multi-word text.
- **Collimate algorithm**: Uses greedy parent-child traversal to follow text flow, making it effective for text that isn't axis-aligned where traditional bucket-based line grouping fails.

## Type of Change

- New feature (non-breaking change that adds functionality)

## Testing

- [x] I have tested this change locally
- [x] I have added/updated tests for this change

**Test details:**
- Added comprehensive unit tests in `test_stitch_ocr_detections_v2.py` covering all three algorithms (tolerance, otsu, collimate)
- Added integration tests in `test_workflow_with_ocr_detections_stitching_v2.py` for workflow execution with each algorithm
- Tests cover empty detections, single/multiple lines, all reading directions, delimiter handling, and algorithm-specific features (skewed text for collimate, bimodality detection for Otsu)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code where necessary, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [x] I have updated the documentation accordingly (if applicable)

### New Otsu Algorithm
- Uses Otsu's method on normalized gaps to find natural word breaks
- Resolution-invariant: gaps normalized by local character width
- Bimodality detection prevents false word breaks in single-word text
- New parameter: `otsu_threshold_multiplier` to adjust threshold sensitivity

### New Collimate Algorithm
- Uses greedy parent-child traversal to build lines by following text flow
- Best for skewed, curved, or non-axis-aligned text
- Fixed bugs from original implementation: vertical orientation (was using height instead of width) and word spacing (was inserting spaces between every character)
- New parameter: `collimate_tolerance` controls alignment tolerance in pixels

### Documentation
- Updated LONG_DESCRIPTION to document all three algorithms with use cases and parameter descriptions

---

> [!NOTE]
> Adds a new `stitch_ocr_detections@v2` block with multiple stitching strategies and updates workflows/tests to use it.
> 
> - Implements `StitchOCRDetectionsBlockV2` supporting `tolerance`, `otsu` (normalized-gap Otsu with bimodality detection, `otsu_threshold_multiplier`), and `collimate` (skew-aware traversal, `collimate_tolerance`); includes `delimiter`, reading-direction handling (including `auto` via helper), and `tolerance` validation
> - Replaces loader registration from `stitch_ocr_detections@v1` to `@v2` and updates integration tests/workflows to reference `@v2`
> - Adds extensive unit/integration tests covering all algorithms, directions, delimiters, and edge cases
> - Adds `visualize_ocr_gaps.py` utility to inspect normalized gaps and Otsu thresholding
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 485ff97fe7169ffd5b9476112aa6f8d9a551dc04. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: inference/core/workflows/core_steps/transformations/stitch_ocr_detections/v2.py [hunk 1]
```diff
@@ -0,0 +1,885 @@
...
+
+class StitchOCRDetectionsBlockV2(WorkflowBlock):
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return BlockManifest
+
+    def run(
+        self,
+        predictions: Batch[sv.Detections],
+        stitching_algorithm: str,
```
Comment (grzegorz-roboflow) [inline inference/core/workflows/core_steps/transformations/stitch_ocr_detections/v2.py]: in block manifest, are we not declaring "tolerance" as default for this param?

File: visualize_ocr_gaps.py [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+        print(f"Reports saved to: {output_base_dir}/")
+
+    else:
+        print("No debug files found in /data/ocr_debug/")
+        print("\nRun the OCR workflow first to generate debug files.")
+        print("\nExpected structure:")
+        print("  /data/ocr_debug/ocr_detections_<uuid>.json")
+        print("\nOutput will be saved to:")
+        print("  /data/ocr_reports/<uuid>/ocr_gap_analysis.png")
+        print("  /data/ocr_reports/<uuid>/ocr_gap_report.json")
```
[the remark this answers is not in the document]
Comment (grzegorz-roboflow) [reply visualize_ocr_gaps.py]: please move this script somewhere else, maybe to examples?

File: inference/core/workflows/core_steps/transformations/stitch_ocr_detections/v2.py [hunk 3]
```diff
@@ -0,0 +1,885 @@
...
+                    "name": "Otsu Thresholding",
+                    "description": "Resolution-invariant algorithm using normalized gaps. Works well with varying font sizes.",
+                },
+                "collimate": {
+                    "name": "Collimate (Skewed Text)",
+                    "description": "Greedy parent-child traversal for grouping. Best for skewed or curved text.",
+                },
+            }
+        },
+    )
```
[the remark this answers is not in the document]
Comment (reiffd7) [reply inference/core/workflows/core_steps/transformations/stitch_ocr_detections/v2.py:135]: added default stitching algorithm of "tolerance"

File: visualize_ocr_gaps.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (reiffd7) [reply visualize_ocr_gaps.py]: removed the script
